src/protocols/LoRaWAN/LoRaWAN.cpp

```diff
@@ -2296,6 +2751,17 @@ void LoRaWANNode::processAES(uint8_t* in, size_t len, uint8_t* key, uint8_t* out
   }
 }
 
+uint16_t LoRaWANNode::checkSum16(uint8_t *key, uint8_t keyLen) {
+  uint16_t buf16[RADIOLIB_AES128_KEY_SIZE/2] = { 0 };
+  uint8_t bufLen = keyLen / 2;
+  memcpy(buf16, key, keyLen);
```

Comment:
One more thing here, since we are copying `keyLen` bytes into a `RADIOLIB_AES128_KEY_SIZE` bytes long array, we should check that `keyLen <= RADIOLIB_AES128_KEY_SIZE` and return or stop at `RADIOLIB_AES128_KEY_SIZE` bytes copied.